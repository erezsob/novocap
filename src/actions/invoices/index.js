// @flow
import queryString from 'query-string';
import * as types from '../../common/ActionTypes';
import { invoiceService } from '../../services/services';
import { handleServerRequestError } from './../app';
import { getVenueFromStorage } from './../../util/venue';

import type { Location } from 'react-router-dom';
import type {
  Invoices,
  Dispatch,
  FetchInvoicesParams,
  SessionVenue
} from './../../common/types';

export function fetchInvoicesSuccess(invoices: Invoices) {
  return (dispatch: Dispatch) => {
    dispatch({ type: types.LOADING, loading: false });
    dispatch({
      type: types.FETCH_INVOICES_SUCCESS,
      invoices
    });
  };
}

export function fetchInvoices(paramsObj: FetchInvoicesParams = {}) {
  return (dispatch: Dispatch) => {
    dispatch({ type: types.FETCHING_INVOICES });
    dispatch({ type: types.LOADING, loading: true });

    const venue: SessionVenue = getVenueFromStorage();

    const stringParameters = queryString.stringify(paramsObj);
    return invoiceService(venue.id, stringParameters)
      .then(response => response.json())
      .then(invoices => dispatch(fetchInvoicesSuccess(invoices)))
      .catch(response => dispatch(handleServerRequestError(response)));
  };
}

export function updateInvoicesResult(location: Location) {
  const queryParams = queryString.parse(location.search);
  const { page, size } = queryParams;
  return (dispatch: Dispatch) => {
    dispatch(fetchInvoices({ page, size }));
  };
}
