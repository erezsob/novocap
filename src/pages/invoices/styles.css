@value bpSMedium from './../../common/variables.css';

.controls {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    margin-top: 20px;
}

.controls > ul {
    padding: 20px 0 10px;
}

.controlsTop {
    composes: controls;
    justify-content: flex-end;
    margin: 0 0 10px;
}

.controlsTop > ul {
    padding: 10px 0;
}

@media(min-width: bpSMedium) and (min-device-width: bpSMedium) {
    .controls {
        height: 40px;
        flex-direction: row;
        padding: 20px 0;
        margin: 0;
    }

    .controls > * {
        padding: 0;
    }

    .controlsTop > ul {
        margin-bottom: 0;
    }
}
