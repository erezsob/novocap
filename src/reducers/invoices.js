import { FETCH_INVOICES_SUCCESS } from '../common/ActionTypes';

const initialState = {
  page: 1,
  size: 20,
  totalElements: undefined,
  totalPages: undefined,
  elements: [
    {
      uuid: '',
      createdAt: '',
      closedAt: '',
      totalAmount: '',
      invoiceNumber: '',
      table: {
        name: '',
        oid: {
          uuid: '',
          version: undefined
        }
      },
      user: {
        name: '',
        oid: {
          uuid: '',
          version: undefined
        }
      },
      paymentTypes: {
        name: '',
        oid: {
          uuid: '',
          version: undefined
        }
      }
    }
  ]
};

export default function invoices(state = initialState, action) {
  switch (action.type) {
    case FETCH_INVOICES_SUCCESS: {
      return { ...state, ...action.invoices };
    }
    default: {
      return state;
    }
  }
}
