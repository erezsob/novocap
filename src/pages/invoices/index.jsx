// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { Redirect } from 'react-router-dom';
import Grid from 'material-ui/Grid';
import { INVOICES } from './../../routes';
import { WIDGETS_CONTAINER_ID } from './../../constants';
import { InvoicesGrid } from '../../containers';
import { updateInvoicesResult } from './../../actions/invoices';
import { Section, ItemsPerPage, Paginator } from './../../components';
import styles from './styles.css';

import type { Invoices as InvoicesType } from './../../common/types';

type Props = {
  pages: number,
  totalPages: number,
  size: number,
  page: number,
  invoices: InvoicesType,
  updateInvoicesResult: Function,
  location: {
    search: string
  }
};

export class Invoices extends Component<Props> {
  props: Props;
  document: Document;

  componentDidMount() {
    const { updateInvoicesResult, location } = this.props;
    updateInvoicesResult(location);
  }

  componentWillReceiveProps(nextProps: Props) {
    const { updateInvoicesResult, location } = this.props;
    const nextLocation = nextProps.location;
    if (nextLocation.search !== location.search) {
      updateInvoicesResult(nextLocation);
    }
  }

  componentDidUpdate(prevProps: Props) {
    if (this.props.location.search !== prevProps.location.search) {
      this.scrollToTop();
    }
  }

  scrollToTop = () => {
    let widgetsContainer = document.getElementById(WIDGETS_CONTAINER_ID);
    if (widgetsContainer instanceof HTMLElement) {
      widgetsContainer.scrollTop = 0;
    }
  };

  renderPagination = (page: number, totalPages: number, size: number) => (
    <Paginator
      currentPage={page}
      totalPages={totalPages}
      pageSize={size}
      baseRoute={INVOICES}
    />
  );

  render() {
    const { invoices, page, totalPages, size } = this.props;
    return (
      <div>
        <Section title="Invoices">
          <Grid item xs={12}>
            <div className={styles.controlsTop}>
              {this.renderPagination(page, totalPages, size)}
            </div>
            <InvoicesGrid invoices={invoices} />
            <div className={styles.controls}>
              <ItemsPerPage size={size} route={INVOICES} />
              {this.renderPagination(page, totalPages, size)}
            </div>
          </Grid>
        </Section>
        {page > totalPages && (
          <Redirect to={`${INVOICES}?page=${totalPages}&size=${size}`} />
        )}
        {page < 1 && <Redirect to={`${INVOICES}?page=1&size=${size}`} />}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  invoices: state.invoices,
  page: state.invoices.page,
  totalPages: state.invoices.totalPages,
  size: state.invoices.size
});

const mapDispatchToProps = {
  updateInvoicesResult
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Invoices)
);
