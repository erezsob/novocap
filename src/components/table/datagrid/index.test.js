import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ReactTable from 'react-table';
import invoices from '../../../../mockdata/invoices.json';
import Datagrid from './index';

describe('<Datagrid />', () => {
  let wrapper;
  let table;
  const columns = [
    {
      Header: 'Invoice Number',
      accessor: 'invoiceNumber'
    },
    {
      id: 'table',
      Header: 'Table',
      accessor: 'table.name'
    },
    {
      id: 'user',
      Header: 'User',
      accessor: 'user.name'
    }
  ];

  beforeEach(() => {
    wrapper = mount(<Datagrid data={invoices.elements} columns={columns} />);
    table = wrapper.find(ReactTable);
  });

  it('renders', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('renders properly', () => {
    const tree = renderer
      .create(<Datagrid data={invoices.elements} columns={columns} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a table with 3 columns', () => {
    expect(table).toHaveLength(1);
    expect(table.find('.rt-thead .rt-tr').children()).toHaveLength(3);
  });

  it('renders as many rows as elements in the invoices array', () => {
    expect(table.find('.rt-tbody').children()).toHaveLength(invoices.size);
  });

  it('shows a message when there is no data to display in the table', () => {
    wrapper = mount(<Datagrid data={[]} columns={columns} />);
    expect(wrapper.find('.rt-noData')).toHaveLength(1);
    expect(wrapper.find('.rt-noData').text()).toBe('No rows found');
  });
});
