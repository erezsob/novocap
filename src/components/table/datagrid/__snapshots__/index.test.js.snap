// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Datagrid /> renders properly 1`] = `
<div
  className="ReactTable -striped"
  style={Object {}}
>
  <div
    className="rt-table"
    role="grid"
    style={undefined}
  >
    <div
      className="rt-thead -header"
      style={
        Object {
          "minWidth": "300px",
        }
      }
    >
      <div
        className="rt-tr"
        role="row"
        style={undefined}
      >
        <div
          className="rt-th"
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "flex": "100 0 auto",
              "maxWidth": null,
              "width": "100px",
            }
          }
          tabIndex="-1"
        >
          <div
            className=""
          >
            Invoice Number
          </div>
        </div>
        <div
          className="rt-th"
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "flex": "100 0 auto",
              "maxWidth": null,
              "width": "100px",
            }
          }
          tabIndex="-1"
        >
          <div
            className=""
          >
            Table
          </div>
        </div>
        <div
          className="rt-th"
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "flex": "100 0 auto",
              "maxWidth": null,
              "width": "100px",
            }
          }
          tabIndex="-1"
        >
          <div
            className=""
          >
            User
          </div>
        </div>
      </div>
    </div>
    <div
      className="rt-tbody"
      style={
        Object {
          "minWidth": "300px",
        }
      }
    >
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            150-0
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 1
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Waitress Charlotte
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            150-12
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 1
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Waitress Katy
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            150-3
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 1
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Manager Javier
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            150-5
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 1
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Manager Javier
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            150-7
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Counter
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Waitress Katy
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            150-6
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 1
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Waitress Charlotte
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            150-4
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 1
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Manager Johnny
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            150-11
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Counter
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Manager Javier
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            150-14
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 2
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Manager Aletta
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            150-10
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 3
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Waitress Katy
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            149-2
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 2
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Manager Javier
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            149-6
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 3
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Waitress Charlotte
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            149-1
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 3
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Manager Javier
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            149-17
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 2
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Waitress Katy
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            149-10
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 1
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Manager Aletta
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            149-13
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 2
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Waitress Katy
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            149-7
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 3
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Waitress Katy
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            149-3
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 1 old
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Manager Javier
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            149-0
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Table 2
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Chef Vincent
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            149-5
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Counter
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            Manager Javier
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="-loading"
  >
    <div
      className="-loading-inner"
    >
      Loading...
    </div>
  </div>
</div>
`;
